Cartesian_kernel/include/CGAL/constructions/kernel_ftC2.h

```diff
@@ -283,9 +283,17 @@ line_get_pointC2(const FT &a, const FT &b, const FT &c, const FT &i,
       // The test `!boost::is_integral<FT>::value` is there to avoid
       // that `a != 0` is tested on anything but integral types, for
       // performance reasons.
-      CGAL_assume(!boost::is_integral<FT>::value || a != FT(0));
-
+      //CGAL_assume(!boost::is_integral<FT>::value || a != FT(0));
+    
+    //the previous code doesn't work, the warning is still there. Definitely
+    //mute it using pragma.
+#ifdef _MSC_VER
+#pragma warning(disable:4723) 
+#endif _MSC_VER
       x = -c/a;
+#ifdef _MSC_VER
+#pragma warning(default:4723)
+#endif _MSC_VER
```

Comment:
That is not correct. If the default is "On" but the users have disabled the warning, the those lines will re-enable the warning! You have to use:
```c++
#pragma warning(push)
#pragma warning(disable:4723)
```
and then:
```c++
#pragma warning(pop)
```

Actually, as the warning is only for one line, it seems that one can do it in one line:
```c++
#ifdef _MSC_VER
#pragma warning(suppress:4723)
#endif
```

See https://docs.microsoft.com/en-us/cpp/preprocessor/warning?view=vs-2015.